Src/AutoFixture/Kernel/FactoryMethodQuery.cs

```diff
@@ -39,6 +43,7 @@ public IEnumerable<IMethod> SelectMethods(Type type)
             return from mi in type.GetMethods(BindingFlags.Static | BindingFlags.Public)
                    where mi.ReturnType == type
                    let parameters = mi.GetParameters()
+                   where mi.GetParameters().All(p => p.ParameterType != type)

```

Comment:
This makes the exception message more general though:

```
Ploeh.AutoFixture.ObjectCreationException : AutoFixture was unable to create an
instance from Ploeh.TestTypeFoundation.TypeWithPseudoFactoryMethod, most likely
because it has no public constructor, is an abstract or non-public type.
```

Compared to the current one:

```
Ploeh.AutoFixture.ObjectCreationException : AutoFixture was unable to create an
instance of type Ploeh.AutoFixture.Kernel.SeededRequest because the traversed
object graph contains a circular reference. Information about the circular path
follows below. This is the correct behavior when a Fixture is equipped with a
ThrowingRecursionBehavior, which is the default. This ensures that you are being
made aware of circular references in your code. Your first reaction should be to
redesign your API in order to get rid of all circular references. However, if
this is not possible (most likely because parts or all of the API is delivered
by a third party), you can replace this default behavior with a different
behavior: on the Fixture instance, remove the ThrowingRecursionBehavior from
Fixture.Behaviors, and instead add an instance of OmitOnRecursionBehavior.

    Path:
        Ploeh.TestTypeFoundation.TypeWithPseudoFactoryMethod argument --> 
        Ploeh.TestTypeFoundation.TypeWithPseudoFactoryMethod --> 
        Ploeh.TestTypeFoundation.TypeWithPseudoFactoryMethod
```